qcodes/instrument_drivers/devices.py

```diff
@@ -1,6 +1,190 @@
 from typing import Union, cast
+import enum
+import warnings
 
-from qcodes import Parameter, Instrument
+from qcodes import Parameter, ManualParameter, Instrument
+
+
+class Role(enum.Enum):
+    GAIN = enum.auto()
+    DIVISION = enum.auto()
+
+
+class ParameterScaler(Parameter):
+    """
+    Parameter Scaler
+
+    To be used when you use a physical voltage divider or an amplifier to set
+    or get a quantity.
+
+    Initialize the parameter by passing the parameter to be measured/set
+    and the value of the division OR the gain.
+
+    The scaling value can be either a scalar value or a Qcodes Parameter.
+
+    The parameter scaler acts a your original parameter, but will set the right
+    value, and store the gain/division in the metadata.
+
+    Examples:
+        Resistive voltage divider
+        >>> vd = ParameterScaler(dac.chan0, division = 10)
+
+        Voltage multiplier
+        >>> vb = ParameterScaler(dac.chan0, gain = 30, name = 'Vb')
+
+        Transimpedance amplifier
+        >>> Id = ParameterScaler(multimeter.amplitude, division = 1e6, name = 'Id', unit = 'A')
+
+    Args:
+        output: Physical Parameter that need conversion
+        division: the division value
+        gain: the gain value
+        label: label of this parameter, by default uses 'output' label
+            but attaches _amplified or _attenuated depending if gain
+            or division has been specified
+        name: name of this parameter, by default uses 'output' name
+            but attaches _amplified or _attenuated depending if gain
+            or division has been specified
+        unit: resulting unit. It uses the one of 'output' by default
+    """
+
+    def __init__(self,
+                 output: Parameter,
+                 division: Union[int, float, Parameter] = None,
+                 gain: Union[int, float, Parameter] = None,
+                 name: str=None,
+                 label: str=None,
+                 unit: str=None) -> None:
+        self._wrapper_param = output
```

Comment:
Could you please rename ‘_wrapper_parameter’ to ‘_wrapped_parameter’? The reason is that scaler wraps the output, hence the output is wrapped inside the scaler, and the scaler is the wrapper here.